lib/src/models/ledger_signing_path.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ledger_cardano/src/utils/constants.dart';
+part 'ledger_signing_path.freezed.dart';
+
+@freezed
+sealed class LedgerSigningPath with _$LedgerSigningPath {
+  LedgerSigningPath._();
+
+  factory LedgerSigningPath.byron({
+    required int account,
+    required int address,
+  }) = LedgerSigningPath_Byron;
+  factory LedgerSigningPath.shelley({
+    required int account,
+    required int address,
+    required ShelleyAddressRole role,
+  }) = LedgerSigningPath_Shelley;
+  factory LedgerSigningPath.custom(List<int> path) = LedgerSigningPath_Custom;
+
+  late final List<int> signingPath = switch (this) {
+    LedgerSigningPath_Byron(account: final account, address: final address) => [
+        harden + 44,
+        harden + 1815,
+        harden + account,
+        0,
+        address
+      ],
+    LedgerSigningPath_Shelley(account: final account, role: ShelleyAddressRole role, address: final address) => [
+        harden + 1852,
+        harden + 1815,
+        harden + account,
+        role.index,
```

Comment:
please explicitly define `derivationIndex` for the `ShelleyAddressRole` enum and use that instead of the auto-generated `index` from enum. I feel like using `index` is a bit risky long-term (cause a change in order in the enum items will break the implementation).